test/integration/query-interface.test.js

```diff
@@ -112,6 +113,114 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
+
+          async function _createSchemaAndTables(schemaName) {
+            await this.sequelize.createSchema(schemaName);
+            await Reflect.apply(createTestTablesForSchema, this, [schemaName]);
+          }
+
+          await Promise.all(baseTestSchemas.map(_createSchemaAndTables.bind(this)));
+        }
+
+        const normalizeTableNames = (tableNames = []) => {
+          if (tableNames[0] && tableNames[0].tableName) {
+            return tableNames.map(v => v.tableName).sort();
+          }
+
+          return tableNames.sort();
+        };
+
+        it('shows all tables from the specified schema in the method options', async function () {
+          await createSchemaAndTables.apply(this);
+          const schemaOneTables = normalizeTableNames(await this.queryInterface.showAllTables({ schema: 'schema_1' }));
+          const schemaTwoTables = normalizeTableNames(await this.queryInterface.showAllTables({ schema: 'schema_2' }));
+
+          expect(schemaOneTables).to.include('schema_1_table_1');
+          expect(schemaOneTables).to.include('schema_1_table_2');
+
+          expect(schemaTwoTables).to.include('schema_2_table_1');
+          expect(schemaTwoTables).to.include('schema_2_table_2');
+        });
+
+        if (dialectName === 'postgres') {
+          it('uses the schema from options instead of initialization options', async () => {
+            const { sequelize } = await getSequelizeInstanceWithSchema();
+            const  schemaThreeTables = normalizeTableNames(await sequelize.queryInterface.showAllTables());
+            const schemaFourTables = normalizeTableNames(await sequelize.queryInterface.showAllTables({ schema: 'schema_4' }));
+
+            expect(schemaThreeTables).to.include('schema_3_table_1');
+            expect(schemaThreeTables).to.include('schema_3_table_2');
+
+            expect(schemaFourTables).to.include('schema_4_table_1');
+            expect(schemaFourTables).to.include('schema_4_table_2');
+          });
+
+          it('defaults to the database name as the schema if it is not specified in init and method options', async function () {
+            await createSchemaAndTables.apply(this);
+            const schemaOneTables = normalizeTableNames(await this.queryInterface.showAllTables());
+            const schemaTwoTables = normalizeTableNames(await this.queryInterface.showAllTables({ schema: 'schema_2' }));
+
+            expect(schemaOneTables).to.include('sequelize_test_table_1');
+            expect(schemaOneTables).to.include('sequelize_test_table_2');
+
+            expect(schemaTwoTables).to.include('schema_2_table_1');
+            expect(schemaTwoTables).to.include('schema_2_table_2');
+          });
+        }
+
+        if (['mariadb', 'mysql'].includes(dialectName)) {
+          it('should show tables from non technical schemas even if the schema is not specified in method options', async function () {
```

Comment:
I'm not sure I understand what this test does

Reply:
I was fixated on the additional clause that's [there](https://github.com/sequelize/sequelize/blob/main/src/dialects/mysql/query-generator.js#L178) for mysql.

Reply:
Ah I see, I think that should be the case for all dialects